timesketch/lib/analyzers/chain_plugins/interface.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+"""This file contains an interface for chain analyzer plugins."""
+import abc
+
+from timesketch.lib import emojis
+
+
+class BaseChainPlugin(object):
+    """A base plugin for the chain analyzer.
+
+    This is an interface for the chain analyzer plugins.
+
+    """
+    NAME = 'chain'
+    DESCRIPTION = ''
+
+    # A string value that defines the search query used to find the original
+    # event that starts the chain. In order for this plugin to work
+    # either the SEARCH_QUERY or SEARCH_QUERY_DSL needs to be defined.
+    SEARCH_QUERY = ''
+
+    # Defines the original event search query DSL. If this attribute
+    # is defined the SEARCH_QUERY attribute is ignored.
+    SEARCH_QUERY_DSL = ''
+
+    # Defines the fields that need to be returned as part of the
+    # event object.
+    EVENT_FIELDS = []
+
+    _EMOJIS = [emojis.get_emoji('LINK')]
+
+    def __init__(self, analyzer_object):
+        """Initialize the plugin."""
+        super(BaseChainPlugin, self).__init__()
+        self.analyzer_object = analyzer_object
+
+    def ProcessChain(self, base_event):
```

Comment:
We are using pep8 style, so this should be:
```
def process_chain()
```

This is throughout this PR as well

Reply:
done... some shameless copy/paste and falling back to old plaso styles ;)